pkg/scheduler/framework/interface.go

```diff
@@ -47,12 +47,24 @@ type NodeScore struct {
 	Score int64
 }
 
-// PluginToNodeScores declares a map from plugin name to its NodeScoreList.
-type PluginToNodeScores map[string]NodeScoreList
-
 // NodeToStatusMap declares map from node name to its status.
 type NodeToStatusMap map[string]*Status
 
+// NodePluginScores is a struct with node name and scores for that node.
+type NodePluginScores struct {
```

Comment:
Maybe we also need a `totalScore/finalScore` here. 
If we want to provide this information to users subsequently, we can't let them do the calculation themselves, WDYT?

Reply:
To add calculated score here totally makes sense 👍 
(But, what is the difference between totalscore and finalscore? (Or did you mean to include either of those?))
I changed the PR to include `totalScore` field and calculate `totalScore` in RunScorePlugin.

I can think of several advantages of your proposed way compared to the current implementation of this PR.
- More efficient. We don't need to calculate total scores in `prioritizeNodes` with loop. Calculations of total score can be performed with existing parallel processing.
- we just need to change `prioritizeNodes` to return `([]framework.NodePluginScores, error)` in the future. (instead of `(framework.NodeScoreList, []framework.NodePluginScores, error)` https://github.com/kubernetes/kubernetes/pull/108534) That's simpler.
- And, by having the total score and the detailed score in the same data (`[]framework.NodePluginScores`), we can reduce the calculation in the future implementation of https://github.com/kubernetes/kubernetes/issues/107556. Once the top M Nodes have been determined, there will be no longer a need to search for detailed scores for those Nodes.

Reply:
> did you mean to include either of those

Yes. Just either of them. 😄